### DISCUSSION TITLE
V2's Cognitive Dissonance

### DISCUSSION THREAD

**Discussion by ohenepee** [WalletConnect/walletconnect-monorepo#2543] [Developer Support]
Personally I feel there's an unnecessary rush with Web3Modal v2 (or new API, etc). For a start there several libraries with no proper definition for each's use, not on GitHub or NPM nor the Docs Website. The examples leave more questions to be asked than the help its suppose to provide.

1. What's the difference between `@web3modal/html` and `@web3modal/standalone`, in which instance is each used?
2. The examples doesn't show how to integrate `ethers.js` or `web3.js` nor the brand new `viem` on which its indirectly built, which makes one wonder where do I start to interact with smart contracts or **how to disconnect the wallet** in Vanilla JavaScript?

**Comment by xzilja** [comment]
Docs have been updated to improve on this https://docs.walletconnect.com/2.0/web/about-web3modal

There are clear paths for html or react now, as well as separation into ethereum libraries and ones you can use "multichain". In terms of ethereum you can choose to either use simple approach with `eip-1193` compliant provider that you can pass to ethers, web3js, viem etc. or go into more advanced route and use it with wagmi, this unlocks some extra features.

Do these answer questions above or are we still missing something?

**Reply by ohenepee** [reply]
I used this particular [guide](https://docs.walletconnect.com/2.0/web/web3modal/html/wagmi/installation), how do I integrate `viem` to **write** to smart contracts and how do I **disconnect the wallet** using Vanilla JavaScript? Btw I've looked at `viem`'s docs but I want to know how to get the `walletClient` in/using the linked guide.

**Reply by xzilja** [reply]
Guide you linked is for wagmi, if you want to use viem you need to follow [ethereum-provider guide](https://docs.walletconnect.com/2.0/web/web3modal/html/ethereum-provider/installation)

You can then pass this provider to viem as a wallet client (https://viem.sh/docs/clients/wallet.html)

Afterwards, you use viem as you'd usually do.

**Reply by jxom** [reply]
Here is an example of WalletConnect v2 + viem: https://stackblitz.com/edit/viem-getting-started-a3ekcn?file=index.tsx (example won't work without a valid WalletConnect Project ID).

**Reply by ohenepee** [reply]
So I've been trying to use the `Ethereum-Provider` with `viem` but as I said previously documentation is spotty and one has to figure out things. In that package's guide, there is a _"chains"_ field in the instantiation options, what kind of value is that field expecting? In all the examples, it is given `[1]` what does that mean or do?

I did this and desktop wallets aren't showing when the modal opens:
```js
import { okc } from "viem/chains";

let ethereumProvider = await EthereumProvider.init({
  projectId: "<my-id-goes-here>",
  showQrModal: true,
  chains: [okc],
  explorerRecommendedWalletIds: [
    "20459438007b75f4f4acb98bf29aa3b800550309646d375da5fd4aac6c2a2c66", // tokenpocket
    "38f5d18bd8522c244bdd70cb4a68e0e718865155811c043f052fb9f1c51de662", // bitkeep
    "dceb063851b1833cbb209e3717a0a0b06bf3fb500fe9db8cd3a553e4b1d02137" // onto
  ]
});
```

**Reply by glitch-txs** [reply]
you should add the chain id which is a number, you can add more than one if want to support multiple chains.
